### DISCUSSION TITLE
Oxygen Generator needs special fuel rods

### DISCUSSION THREAD

**Discussion by Veritius** [FakeFishGames/Barotrauma#6210] [Ideas and suggestions]
The Oxygen Generator needs specialised "fuel rods" (not the reactor kind) storing oxygen chemically. Extremely slow-burning, and with an alarm when the rod is out.
This is how actual submarines generate their oxygen, with chemical rods.

Also, Oxygenite Shards would have a purpose other than Liquid Oxygenite.

**Comment by MateusJK15** [comment]
IRL submarines also generates oxygen from sea water by using chemical reactions to remove salt from the water and to separate hydrogen from oxygen, so the "oxygen rods" (they really exists, but not with this name) are used only as an emergency oxygen source.

**Reply by Veritius** [reply]
This is a much better idea. If the reactor was off, the oxygen generator could take specialised emergency oxygen rods that put air through the vents (as long as there's no water)

**Reply by MateusJK15** [reply]
If you want to know exactly how it works : httpss://youtu.be/g3Ud6mHdhlQ 
The guy in this video went aboard a military submarine and with the help of the crew, made an awesome explanation about how these systems works.

**Reply by Stallfighter** [reply]
> 
> 
> This is a much better idea. If the reactor was off, the oxygen generator could take specialised emergency oxygen rods that put air through the vents (as long as there's no water)

Yes, and how exactly you are going to use (I'll call it as it is) sodium/lithium peroxide filters when there are no power, which means that ventilation system is out? The only thing that will work in that situation is chlorate candle, and ONLY inside its compartment.
